modules/system/models/MailTemplate.php

```diff
@@ -161,11 +164,15 @@ public static function addContentToMailer($message, $code, $data)
         /*
          * HTML contents
          */
+        if (preg_match('{{(\s*message*\s*)}}', $template->content_html)) {
+            throw new ApplicationException('Do not use the \'message\' variable.');
```

Comment:
This is not a very good solution, it's a bandaid at best. Simply telling people that they can't use `{{ message }}` is not good enough, we'll need to better examine the problem and come up with a more robust solution that properly isolates the Message object from being used as a variable within the template.